File: iva.py
import math

import psycopg2
from tabulate import tabulate
def  post(x,y,z):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO ivas(precio, iva, total) VALUES (%s, %s, %s);", (x, y, z))
    conexion.commit()
    cursor.close()
    conexion.close()

def ret():
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
    
    cursor = conexion.cursor()
    cursor.execute("SELECT * from ivas;")
    print(tabulate(cursor, headers=["ide", "precio", "iva", "total"], tablefmt="orgtbl", numalign ="center"))


while True:
    try:
        print("Elija una opcion: 1. Calcular iva y total del producto,   2. Historial,  3.Salir: ")
        x=int(input("Opcion a elegir: \n"))
        if x>3 or x<=0:
            print("Numero erroneo, vuelva a ingresar: \n")
        elif x==1:
            precio=float(input("Ingrese el precio del producto: "))
            iva= precio*0.12
            print("El IVA en el producto es de: ", iva)
            total=precio+iva
            print("El precio total es de: ", total)
            post(precio, iva, total)
        elif x==2:
            ret()
            break
        elif x==3:
            break

    except Exception as es:
        print("\n Ingreso un caracter y no un numero, vuelva a intentar:\n")
        print(repr(es))

exit(0)

File: iva.m
pkg load database;
clc; 
clear all; 
close all;

retry= true;
while retry
  try
    fprintf("Elija una opcion: 1. Calcular iva y total del producto,   2. Historial)");
    i=(input ("\n Opcion a elegir: "));
    if i>3||i<0
      fprintf("\n Numero erroneo, vuelva a ingresar: ");
    endif
      if i==1
        precio=input("Ingrese el precio del producto: ")
        iva= precio*0.12;
        fprintf("El IVA en el producto es de: %d \n", iva)
        total=precio+iva;
        fprintf("El precio total es de: %d \n", total)
        %Conectar con PGADMIN
        conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
        'port','5432','user','postgres','password','jblue5queso'));
        params=cell(1,3);
        params{1,1}=precio;
        params{1,2}=iva;
        params{1,3}=total;
        N=pq_exec_params(conn, "insert into ivas(precio, iva, total) values($1,$2,$3);",params); %insertar datos en la tabla
        %Texto
        save('ivas.txt','precio', 'iva', 'total','-ascii');

    endif
    if i==2
       conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
       'port','5432','user','postgres','password','jblue5queso'));
       N=pq_exec_params(conn, "Select * from ivas;");%historial de la tabla
       disp(N);
       retry=false;
    endif
   catch
      fprintf("Solo se permiten valores numericos, vuelva a selecionar \n");
      msg = lasterror.message;
      fprintf(msg);
    end_try_catch
 endwhile
 
 